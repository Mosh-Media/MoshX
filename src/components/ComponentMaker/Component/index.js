import React, { Suspense } from "react";
import Components from "../Components"
import Spinner from "../../shared/Spinner";

const Component = props => {
	const Component = Components[props.content];
	return (
		<Suspense fallback={<Spinner loading={true} color="primary" />}>
			<Component>{props.content}</Component>
		</Suspense>
	);
};

export default Component;
