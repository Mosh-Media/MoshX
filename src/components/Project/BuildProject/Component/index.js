import React, { Suspense } from "react";

export const Component = props => {
    const Component = React.lazy(() => import(`../../../../library/components/${props.type}`))
    debugger
	return (
		<Suspense fallback={<div>Loading...</div>}>
			<Component>{props.type}</Component>
		</Suspense>
	);
};
