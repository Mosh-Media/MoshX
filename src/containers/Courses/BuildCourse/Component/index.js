import React, { Suspense, useState } from "react";
import Spinner from "../../../../components/shared/Spinner";

export const Component = props => {
    const Component = React.lazy(() => import(`../../../../library/components/${props.topic}`))
	const [state, setState] = useState({status: ""});
    return (
		<Suspense fallback={<Spinner loading={true} color="primary" />}>
			<Component onClick={() => setState({status: "Saved!"})} {...props}></Component>
			<p>{state.status}</p>
		</Suspense>
	);
};
