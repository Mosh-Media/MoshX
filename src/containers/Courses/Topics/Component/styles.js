const styles = theme => ({
	iframe: {
		borderRadius: theme.shape.borderRadius,
		border: "none"
	},
	card: {
		borderRadius: theme.shape.borderRadius,
		boxShadow: "0 0 51px 0 rgba(0,0,0,.08), 0 6px 18px 0 rgba(0,0,0,.05)",
		"&:hover": {
			boxShadow: "0 0 114px 0 rgba(0,0,0,.08), 0 30px 25px 0 rgba(0,0,0,.05)"
		},
		backgroundColor: "none",
		marginBottom: "20px"
	},
	audioCard: {
		backgroundColor: "#f2f3f4",
		padding: "10px",
		height: "50px"
	},
	playgroundCard: {
		height: "550px"
	}
});

export default styles;
