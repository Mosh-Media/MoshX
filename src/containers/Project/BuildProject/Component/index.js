import React, { Suspense, useState } from "react";

export const Component = props => {
    const Component = React.lazy(() => import(`../../../../library/components/${props.type}`))
	const [state, setState] = useState({status: ""});
    return (
		<Suspense fallback={<div>Loading...</div>}>
			<Component onClick={() => setState({status: "Saved!"})} {...props}></Component>
			<p>{state.status}</p>
		</Suspense>
	);
};
