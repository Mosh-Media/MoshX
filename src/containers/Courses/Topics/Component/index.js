import React, { Suspense, useState } from "react";

import Dnd from "../../../../components/Dnd";
// import ReactPlayer from "react-player";
import Spinner from "../../../../components/shared/Spinner";
import styles from "./styles";
import withStyles from "@material-ui/core/styles/withStyles";

const Component = ({ topics, classes }) => {
	const [state] = useState({ status: "" });
	return (
		<Suspense fallback={<Spinner loading={true} color="primary" />}>
			{/* <div className={`${classes.card} ${classes.audioCard}`}>
				{topics && topics.podcast && (
					<ReactPlayer
						url={topics.podcast}
						playing
						controls={true}
						light={false}
						width="100%"
						height="50px"
					/>
				)}
			</div> */}
			{/* <div
				onClick={() => setState({ status: "Saved!" })}
				className={`${classes.card} ${classes.playgroundCard}`}
			>
				{topics && topics.stackblitz && (
					<iframe
						src={topics.stackblitz}
						title="basic-html-moskool"
						width="100%"
						height="100%"
						className={classes.iframe}
					/>
				)}
			</div> */}
			<Dnd></Dnd>

			<p>{state.status}</p>
		</Suspense>
	);
};

export default withStyles(styles)(Component);