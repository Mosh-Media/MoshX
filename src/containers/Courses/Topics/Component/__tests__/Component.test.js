import React from "react";
import { Component } from "../../Component";
import renderer from "react-test-renderer";

it("Project Component", () => {
	const tree = renderer.create(<Component />).toJSON();
	expect(tree).toMatchSnapshot();
});
