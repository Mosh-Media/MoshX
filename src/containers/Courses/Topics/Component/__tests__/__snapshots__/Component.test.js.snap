// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Component Matches the snapshot 1`] = `
<div>
   Component 
</div>
`;
